File: src/test/scala/com/abaddon83/cardsGames/burracoGames/domainModels/burracoGames/BurracoGameTest.scala
package com.abaddon83.cardsGames.burracoGames.domainModels.burracoGames

import com.abaddon83.cardsGames.burracoGames.domainModels.burracoGames.waitingPlayers.BurracoGameWaitingPlayers
import com.abaddon83.cardsGames.shares.games.GameIdentity
import org.scalatest.funsuite.AnyFunSuite

class BurracoGameTest extends AnyFunSuite{

  test("Given anew game identity, when I create a new Burraco game, then I have a new game") {
    val gameId=GameIdentity()
    val game = BurracoGame.createNewBurracoGame(gameId)
    assert(game.isInstanceOf[BurracoGameWaitingPlayers])
    assert(game.numPlayers == 0)
    assert(game.identity() == gameId)
  }

}

File: src/test/kotlin/com/abaddon83/burracoGame/domainModels/BurracoGameTest.kt
package com.abaddon83.burracoGame.domainModels

import com.abaddon83.burracoGame.writeModel.models.BurracoGame
import com.abaddon83.burracoGame.writeModel.models.games.GameIdentity
import org.junit.Test

class BurracoGameTest {

    @Test
    fun `Given anew game identity, when I create a new Burraco game, then I have a new game`(){
        val gameId= GameIdentity.create()
        val game = BurracoGame.create(gameId)
        assert(game.numPlayers() == 0)
        assert(game.identity() == gameId)
    }
}